kernel-testkit/shared/src/main/scala/cats/effect/kernel/testkit/TestContext.scala

```diff
@@ -130,24 +133,32 @@ final class TestContext private (_seed: Long) extends ExecutionContext { self =>
    *   `true` if a task was available in the internal queue, and was executed, or `false`
    *   otherwise
    */
-  def tickOne(): Boolean =
-    synchronized {
+  def tickOne(): Boolean = {
+    val head = synchronized {
       val current = stateRef
 
-      // extracting one task by taking the immediate tasks
-      extractOneTask(current, current.clock, random) match {
-        case Some((head, rest)) =>
+      // extracing one task
+      current.tasks.headOption.filter(_.runsAt <= current.clock) match {
```

Comment:
Shouldn't we invert this?

```suggestion
      current.tasks.filter(_.runsAt <= current.clock).headOption match {
```

I believe that by doing it in the order you do, you're forcing a lot of churn and and failed takes when most tasks are delayed.

Reply:
Sorry, I don't quite understand this suggestion. I didn't change the code, it was already like that. The code as it is doesn't traverse the data structure to filter it, while the suggestion does, I would think the latter does strictly more work.

Reply:
Oh, I think I missed the fact that `tasks` is sorted, so you're guaranteed to get one that you can run if *any* can be run.